src/plugins/intel_gpu/tests/unit/test_cases/pooling_gpu_test.cpp

```diff
@@ -1517,6 +1517,63 @@ TEST(pooling_forward_gpu, b_fs_yx_fsv4)
+
+    topology topology;
+    topology.add(input_layout("input", input_prim->get_layout()));
+    topology.add(reorder("reorder_input", input_info("input"), layout(data_types::i8, format::fs_b_yx_fsv32, { 2, 32, 1, 2 })));
+    topology.add(pooling("avg_pooling", input_info("reorder_input"), pooling_mode::average, { 7, 7 }, { 2, 2 }));
+    topology.add(reorder("reorder_after_pooling", input_info("avg_pooling"), layout(data_types::f32, format::fs_b_yx_fsv32, { 2, 32, 1, 1 })));
+
+    network network(engine, topology, get_test_default_config(engine));
+    set_values(input_prim, { 41.f, 42.f, 43.f, 44.f, 45.f, 46.f, 47.f, 48.f,
+                             11.f, 12.f, 13.f, 14.f, 15.f, 16.f, 17.f, 18.f,
+                             21.f, 22.f, 23.f, 24.f, 25.f, 26.f, 27.f, 28.f,
+                             31.f, 32.f, 33.f, 34.f, 35.f, 36.f, 37.f, 38.f,
+
+                             31.f, 32.f, 33.f, 34.f, 35.f, 36.f, 37.f, 38.f,
+                             21.f, 22.f, 23.f, 24.f, 25.f, 26.f, 27.f, 28.f,
+                             51.f, 52.f, 53.f, 54.f, 55.f, 56.f, 57.f, 58.f,
+                             11.f, 12.f, 13.f, 14.f, 15.f, 16.f, 17.f, 18.f,
+                             
+                             41.f, 42.f, 43.f, 44.f, 45.f, 46.f, 47.f, 48.f,
+                             11.f, 12.f, 13.f, 14.f, 15.f, 16.f, 17.f, 18.f,
+                             21.f, 22.f, 23.f, 24.f, 25.f, 26.f, 27.f, 28.f,
+                             31.f, 32.f, 33.f, 34.f, 35.f, 36.f, 37.f, 38.f,
+
+                             31.f, 32.f, 33.f, 34.f, 35.f, 36.f, 37.f, 38.f,
+                             21.f, 22.f, 23.f, 24.f, 25.f, 26.f, 27.f, 28.f,
+                             51.f, 52.f, 53.f, 54.f, 55.f, 56.f, 57.f, 58.f,
+                             11.f, 12.f, 13.f, 14.f, 15.f, 16.f, 17.f, 18.f });
+    network.set_input_data("input", input_prim);
+
+    auto outputs = network.execute();
+    ASSERT_EQ(outputs.size(), size_t(1));
+    ASSERT_EQ(outputs.begin()->first, "reorder_after_pooling");
+
+    auto output_prim = outputs.begin()->second.get_memory();
+
+    cldnn::mem_lock<float> output_ptr(output_prim, get_test_stream());
+
+    std::vector<float> ref_data = { 41.f, 43.f, 45.f, 47.f, 11.f, 13.f, 15.f, 17.f,
+                                    21.f, 23.f, 25.f, 27.f, 31.f, 33.f, 35.f, 37.f,
+                                    31.f, 33.f, 35.f, 37.f, 21.f, 23.f, 25.f, 27.f,
+                                    51.f, 53.f, 55.f, 57.f, 11.f, 13.f, 15.f, 17.f,
+
+                                    42.f, 44.f, 46.f, 48.f, 12.f, 14.f, 16.f, 18.f,
+                                    22.f, 24.f, 26.f, 28.f, 32.f, 34.f, 36.f, 38.f,
+                                    32.f, 34.f, 36.f, 38.f, 22.f, 24.f, 26.f, 28.f,
+                                    52.f, 54.f, 56.f, 58.f, 12.f, 14.f, 16.f, 18.f };
+
+    for (size_t i = 0; i < ref_data.size(); i++) {
+        ASSERT_EQ(ref_data[i], float(output_ptr[i]));
+    }
+}
+
```

Comment:
I think we must have similar test for other layouts. It would be better to reuse existing case. or can you generalize the existing case to cover this layout too?

Reply:
What other layout? Why? All I need to implement is "fs_b_yx_fsv32" format and "int8" input type combination.

Reply:
Added more unittests

Reply:
Don't we have similar test case that can cover this case?

Reply:
The added unittests are copied from f16/f32 fs_b_yx_fsv32 cases. If there was any tests to cover int8/f32 fs_b_yx_fsv32 pooling cases, the tests would be failed. 
If you have any cases considering, please let me know.

Reply:
Yes, I've reviewed similar test cases and couldn't find exact target. f16/f32 fs_b_yx_fsv32 target case was most similar case and I implemented from this for int8/uint8/fs_b_yx_fsv32 cases.

Reply:
Added "fs_b_yx_fsv32" format on top of existing low precision test, and deleted my previous tests.